scripts/build.py

```diff
@@ -571,15 +571,18 @@ def build_files(source, target, file_hashes, file_formats=None):
     """If no specific file formats is provided, minifies all CSS and JS files,
     removes whitespace from HTML and interpolates paths in HTML to include
     hashes in source directory and copies it to target.
+    If specific file formats is provided, only files pertaining to the provided
+    formats will be minified.
```

Comment:
Ah. I have comments below about always building HTML in the main function.
So when `build_file(...file_formats=('.html')` is called, only HTML build tasks are queued.
Should I be providing a straightforward example in the docstring?

Reply:
I think the docstring needs to be complete, in terms of explaining what exactly the function does for all possible choices of arguments. As long as that's the case, then it's fine; currently it's not clear what it does to html strings if specific file formats are provided.

Reply:
Now that `build_files` has the ability to pick and choose which type of file formats to be built in this PR, I am using `build_files` to build HTML files first and foremost when an existing staging dir is detected within /oppia. This is the first step.

The 2nd step is I will then filter out list of recently changed files and call `minify_func` directly on those recently changed files, without going through `build_files`. This is because I already know where the files are , using the hash dict, so I do not have to use `build_files` to walk through the directory to find the file.

In retrospect, I feel that the only detail that is missing from the docstring is the fact that there should always be a `build_file(...formats=('.html')` call to build all HTML files for any of directory (/templates or /extensions). Is that clear enough from your perspective?